Float values serialization tests

vitvakatu opened the pull request:
- Added license to the new `float` module
- Fixed some doc tests in `float` module
- Added tests for the following cases: zero, subnormal (denormal) number, normal number, maximum and minimum numbers, +/- infinity, NaNs.

DarkEld3r commented:
Please review: @vldm, @alekseysidorov, @aleksuss, @DenisKolodin, @ozkriff. :bell:

DarkEld3r commented on exonum/src/encoding/tests.rs:
```diff
@@ -144,6 +148,48 @@ fn test_i64_segment() {
     assert_write_check_read(dat, 8);
 }
 
+#[cfg(feature="float_serialize")]
```
Maybe this should be moved into the `float` module too?..

@vldm What do you think?

vitvakatu commented:
I've moved tests to the `float` module and rewritten them a bit.
Also I changed wrapper's constructors in order to forbid subnormal numbers.
Requesting review from @vldm

vitvakatu commented:
Do I need to rebase?

vldm commented on exonum/src/encoding/float.rs:
```diff
@@ -244,3 +258,74 @@ impl ExonumJson for F64 {
+    }
+
+    #[test]
+    fn test_f64_segment() {
```
This test is for f32 value, not segment of f32

vitvakatu commented:
Do not merge for now, needs rebase

DarkEld3r commented on exonum/src/encoding/float.rs:
```diff
@@ -65,10 +66,9 @@ impl F32 {
-            None
+        match value.classify() {
+            FpCategory::Normal | FpCategory::Zero => Some(Self { value }),
+            _ => None,
```
I propose to move this check into a function.

vitvakatu replied:
I don't think it worth it.

DarkEld3r commented on exonum/src/encoding/float.rs:
```diff
@@ -270,7 +269,13 @@ mod tests {
 
-    fn assert_validate_constructor<T, V, C: Fn(V) -> T>(constructor: C, value: V, buffer: &[u8], header_size: Offset)
+    #[derive(Debug, PartialEq)]
+    enum ValidationResult {
```
Is it worth creating an enum for that?

vitvakatu replied:
Will replace with bool
